osquery/tables/system/windows/system_info.cpp

```diff
@@ -149,6 +149,21 @@ QueryData genSystemInfo(QueryContext& context) {
     break;
   }
 
+  // If we're running in ARM x86 emulation, we can get the true processor type
+  // Only available on Win 10 and later
+  typedef BOOL(WINAPI *LPFN_ISWOW64PROCESS2)(HANDLE hProcess, USHORT *pProcessMachine, USHORT *pNativeMachine);
+
+  auto pIsWow64Process2 = reinterpret_cast<LPFN_ISWOW64PROCESS2>(GetProcAddress(GetModuleHandle(L"kernel32.dll"), "IsWow64Process2"));
+  if (pIsWow64Process2 != nullptr) {
+    USHORT pProcessMachine, pNativeMachine;
+
+    if (pIsWow64Process2(GetCurrentProcess(), &pProcessMachine, &pNativeMachine)) {
+      if (pNativeMachine == IMAGE_FILE_MACHINE_ARM64) {
+        r["cpu_type"] = "ARM";
```

Comment:
mmm, this is interesting. I wonder what we actually want to expose here. There's an argument for exposing the emulated one or the underlying one. I wonder if it needs a new column

Reply:
I would assume most people want the correct `cpu_type`?

Reply:
So I sort of understand why you might want the emulated CPU, but my assumption would be that if you're running and x86 osquery binary on an ARM machine, it's because that's what the installer provides by default, and that you want to collect accurate statistics on the machine

Reply:
Yes, maybe. Honestly, I've seen both sides of this

Reply:
Is there another way to tell if your osquery binary is the _wrong_ one for the platform?

Reply:
Perhaps add another column like `emulated_cpu_type`?

Reply:
The emulated CPU is only visible to the process, it doesn't really say anything about the wider system. I can add the column if you think it's the right thing to do though

Reply:
Added!

Reply:
I'm sorta thinking about @directionless' question... Having both `cpu_type` (which I think should be the actual hardware type) and `emulated_cpu_type` (which would be the osquery build type) allows checking whether they match.

Reply:
@dantecatalfamo thank you for adding it! Can you please confirm what results would be observed on eg.
- an ARM machine running an x86 binary
- an ARM machine running an ARM binary
- an x86 machine running an x86 binary
- an x86 machine running an ARM binary (does this happen?)

Reply:
@zwass I actually wasn't able to compile the ARM binary on my system. I followed the build instructions and tried replacing `cmake ... -A x86` with `-A ARM64`, and it generates the right files, but when I go to compile I get errors about an amazon SDK not having the right assembly https://github.com/fleetdm/fleet/issues/26465#issuecomment-2734509655

Reply:
I can confirm that X86 on ARM works as expected though